Source/ALS/Private/AlsCharacter_Actions.cpp

```diff
@@ -903,6 +903,93 @@ void AAlsCharacter::RefreshRagdolling(const float DeltaTime)
 
 		LimitRagdollSpeed();
 	}
+
+	if (IsRagdollingGroundedAndAged())
+	{
+		// Determine whether the ragdoll is facing upward or downward and set the target rotation accordingly.
+
+		const auto PelvisRotation{GetMesh()->GetSocketTransform(UAlsConstants::PelvisBoneName()).Rotator()};
+
+		const auto PelvisDirDotUp{PelvisRotation.RotateVector(FVector::RightVector).Dot(FVector::UpVector)};
+
+		if (RagdollingState.bFacingUpward)
+		{
+			if (PelvisDirDotUp < -0.5f)
+			{
+				RagdollingState.bFacingUpward = false;
+			}
+		}
+		else
+		{
+			if (PelvisDirDotUp > 0.5f)
+			{
+				RagdollingState.bFacingUpward = true;
+			}
+		}
```

Comment:
There were cases where the orientation of the Pelvis bone would occasionally fall right on the boundary of being either upward or downward. This would cause the bFacingUpward flag to flip every few frames, causing the capsule’s orientation to change by 180 degrees each time, resulting in unnatural behavior.
To address this, I made a change so that once it is determined to be facing upward, it will not change the bFacingUpward flag until it is sufficiently facing downward, and vice versa if it is facing downward, it will not change until it is sufficiently facing upward.